src/Morpho.sol

```diff
@@ -377,11 +377,16 @@ contract Morpho is IMorpho {
         uint256 badDebtShares;
         if (position[id][borrower].collateral == 0) {
             badDebtShares = position[id][borrower].borrowShares;
-            uint256 badDebt = badDebtShares.toAssetsUp(market[id].totalBorrowAssets, market[id].totalBorrowShares);
-            market[id].totalSupplyAssets -= badDebt.toUint128();
-            market[id].totalBorrowAssets -= badDebt.toUint128();
-            market[id].totalBorrowShares -= badDebtShares.toUint128();
             position[id][borrower].borrowShares = 0;
+
+            uint128 totalBorrowAssets = market[id].totalBorrowAssets;
+            uint256 badDebt = badDebtShares.toAssetsUp(totalBorrowAssets, market[id].totalBorrowShares);
+            uint128 newTotalBorrowAssets = UtilsLib.zeroFloorSub(totalBorrowAssets, badDebt).toUint128();
+            uint128 newBadDebt = totalBorrowAssets - newTotalBorrowAssets;
```

Comment:
wouldn't it be equivalent but easier to have `badDebt = UtilsLib.min(totalBorrowAssets, badDebtShares.toAssetsUp(totalBorrowAssets, market[id].totalBorrowShares))`?

Reply:
I'll try